archive/tar.go

```diff
@@ -11,7 +11,27 @@ import (
 	"time"
 )
 
-func WriteFilesToTar(dest string, uid, gid int, files ...string) (string, map[string]struct{}, error) {
+type TarWriter interface {
+	WriteHeader(hdr *tar.Header) error
+	Write(b []byte) (int, error)
+	Close() error
+}
+
+type WriterFactory interface {
```

Comment:
IIRC the abstraction is needed because the tar-related helpers that receive this factory sometimes need an image-aware factory and other times need the `defaultTarWriterFactory` (when writing generic tars that aren't going to end up in image layers).

As for further refactors, unless you feel strongly, I think it'd be best to keep the factory simple at first, just to serve the primary goal of this PR.

As for having it do more, I'd prefer to keep this factory focused and simple for now. It may be a good refactor, but the PR is more focused on getting functionality